core/lib/zksync_core/src/lib.rs

```diff
@@ -617,6 +625,19 @@ pub async fn initialize_components(
     .await
     .context("add_witness_generator_to_task_futures()")?;
 
+    if components.contains(&Component::BasicWitnessInputProducer) {
```

Comment:
Uh, and one more. Seems like this component will only run one task at a time, utilizing a single thread (VM is single threaded IIRC). Wouldn't we want to have better pod utilization, running N tasks in parallel?

This is assuming that, like other components, we plan to run these producers as stand-alone components and maybe horizontally scale them.

Reply:
@popzxc but then we can probably just have pods with a single CPU for this?

Reply:
@RomanBrodetski true. I don't know what's more cost-effective, have N single CPU nodes or have N/M M-CPU nodes. If we can get a single CPU with performance similar to the node running state keeper, I guess it would be sufficient.

Reply:
it's just so much simpler to have it single-threaded.. We pay for nodes, not for pods, so if it's cheaper to have a noce with 4 CPUs we can just run 4 pods 1 CPU each there (IIUC)

Reply:
Sure